File: 0000001_twoSum/go/solution_test.go
package twosum_test

import (
	"testing"

	twosum "github.com/XOzymandiasz/leetcode-solutions/0000001_twoSum"
)

func TestTwoSum(t *testing.T) {
	tests := []struct {
		name   string
		nums   []int
		target int
	}{
		{"basic", []int{2, 7, 11, 15}, 9},
		{"unordered", []int{3, 2, 4}, 6},
		{"duplicates", []int{3, 3}, 6},
		{"negative", []int{7, 3, 4, -1}, 2},
	}

	for _, tc := range tests {
		t.Run(tc.name, func(t *testing.T) {
			result := twosum.TwoSum(tc.nums, tc.target)

			i := result[0]
			j := result[1]

			if i == j {
				t.Fatalf("indices must be different: %v", result)
			}

			sum := tc.nums[i] + tc.nums[j]
			if sum != tc.target {
				t.Errorf(
					"nums[%d]=%d + nums[%d]=%d = %d, expected %d",
					i, tc.nums[i],
					j, tc.nums[j],
					sum,
					tc.target,
				)
			}
		})
	}
}

func TestTwoSum_NoSolution(t *testing.T) {
	result := twosum.TwoSum([]int{3}, 100)

	if len(result) != 0 {
		t.Errorf("expected empty result, got %v", result)
	}
}

File: 0000001_twoSum/typescript/solution.test.ts
import { describe, it, expect } from 'vitest';
import { twoSum } from './solution';

describe('twoSum', () => {
    const cases = [
        { name: 'basic', nums: [2, 7, 11, 15], target: 9 },
        { name: 'unordered', nums: [3, 2, 4], target: 6 },
        { name: 'duplicates', nums: [3, 3], target: 6 },
        { name: 'negative', nums: [7, 3, 4, -1], target: 2 },
    ];

    cases.forEach(({ name, nums, target }) => {
        it(name, () => {
            const [i, j] = twoSum(nums, target);
            expect(i).not.toBe(j);
            expect(nums[i] + nums[j]).toBe(target);
        });
    });

    it('returns empty array when no solution exists', () => {
        const result = twoSum([1, 2, 3], 100);
        expect(result).toEqual([]);
    });
});